custom_components/imou_life/__init__.py

```
@@ -245,19 +281,36 @@ async def _setup_coordinator(
         _LOGGER.debug("Fetching initial data with timeout %d seconds...", setup_timeout)
         await asyncio.wait_for(coordinator.async_refresh(), timeout=setup_timeout)
         _LOGGER.debug("Initial data fetch completed")
+
+        # Clear rate limit state on successful data fetch
+        rate_limit_mgr.clear_rate_limit(app_id, app_secret)
+
     except asyncio.TimeoutError:
         _LOGGER.error("Initial data fetch timed out after %d seconds", setup_timeout)
         raise ConfigEntryNotReady(
             translation_domain=DOMAIN, translation_key="initial_fetch_timeout"
         )
```

Ruff (0.15.13)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)